diesel_derives/src/lib.rs

```diff
@@ -234,6 +238,10 @@ pub fn derive_from_sql_row(input: TokenStream) -> TokenStream {
 /// Our rules for inferring table names is considered public API.
 /// It will never change without a major version bump.
 ///
+/// This derive generates the following impls:
+/// * `impl Identifiable for &'a YourType`
+/// * `impl Identifiable for &'a &'a YourType`
```

Comment:
Couple semi-related things:
1/ Does this need to be updated as per previous review's comment?
```suggestion
/// * `impl Identifiable for &'_ &'a YourType`
```

2/ I'm scared of lifetimes but this still seems odd to me; why would one ever need to specify the same lifetime `'a` twice in a row? (Assuming that this constraint _was_ needed?

Reply:
> 2/ I'm scared of lifetimes but this still seems odd to me; why would one ever need to specify the same lifetime 'a twice in a row? (Assuming that this constraint was needed?)

That's a question not that simple to answer. Generally speaking I think different lifetimes for nested references like that one are only relevant if you deal with multiple different reference types at once and one of them is mutable. Otherwise rustc should be able to just use the shortest of all involved lifetimes as "union" of all involved lifetimes. As soon a as mutable reference is involved that becomes tricky, because you cannot hold multiple mutable references to the same object at the same time.

For this specific case: If you have two nested references like that one you can write that as `&'a &'b Type`. From there you can infer that the lifetime `'b` needs to be at least as long as `'a` or to word it differently: `'a` could be a shorter lifetime than `'b'. If now due to some external need `'a` and `'b` are coupled to mutable references this could cause issues if we would require that both are the same lifetime.

EDIT: 

(It the buttons here in the wrong order, so the PR was already merged before I could apply your suggestion :man_facepalming: . Pushed that to 5b9c3bd4c0fb2b63733d2b8baaafce23c8f3ba78)